webapp-django/crashstats/api/views.py

```diff
@@ -114,7 +116,53 @@ def is_valid_model_class(model):
     )
 
 
+def clear_empty_session(fun):
+    """Clears empty sessions so they don't persist"""
+
+    @wraps(fun)
+    def _clear_empty_session(request, *args, **kwargs):
+        # Run the function
+        ret = fun(request, *args, **kwargs)
+
+        # Clear the session if nothing was saved to it
+        session_keys = [
+            key for key in request.session.keys() if not key.startswith("_auth_user")
+        ]
+        if not session_keys:
+            request.session.clear()
+            request.session.flush()
+        return ret
```

Comment:
Because the token middleware logs the "user" in using the token, that creates a session. This code checks to see if the view added anything to the session and if not, clears it. That way, the session never makes it to the db and never goes in the cookie.